File: algo/insertion_sort.py
def insertion_sort(arr):
    for i in range(1, len(arr)):
        j = i - 1
        while j >= 0 and arr[j] > arr[j + 1]:
            arr[j], arr[j + 1] = arr[j + 1], arr[j]
            j -= 1
    return arr

def test_insertion_sort():
    arr = [1, 3, 2, 4, 5]
    insertion_sort(arr)
    assert arr == [1, 2, 3, 4, 5]

    arr = [5, 4, 3, 2, 1]
    insertion_sort(arr)
    assert arr == [1, 2, 3, 4, 5]

    arr = [1, 2, 3, 4, 5]
    insertion_sort(arr)
    assert arr == [1, 2, 3, 4, 5]

    arr = [1]
    insertion_sort(arr)
    assert arr == [1]

    arr = []
    insertion_sort(arr)
    assert arr == []

    print("All tests pass")

test_insertion_sort()

File: algo/insertion_sort.js
function insertionSort(arr){
    for(let i = 1; i < arr.length; i++){
        let j = i - 1;
        while(j >= 0 && arr[j] > arr[j + 1]){
            let tmp = arr[j];
            arr[j] = arr[j + 1];
            arr[j + 1] = tmp;
            j--;
        }
    }
    return arr;
}

function test_insertion_sort(){
    const assert = require('assert');

    let arr = insertionSort([1, 4, 2, 3, 5]);

    assert(arr[0] === 1);
    assert(arr[1] === 2);
    assert(arr[2] === 3);
    assert(arr[3] === 4);
    assert(arr[4] === 5);

    console.log('All test cases pass');
}

test_insertion_sort();